1. Multiples of 3 and 5:

Find the sum of all the multiples of 3 or 5 below 1000

In [1]:
import numpy as np

In [11]:
alist=list(np.arange(1,1001))

In [12]:
ret=[]
def multiples_three_five(somelist):
    for i in alist:
        if (i%3 == 0) or (i%5== 0):
            ret.append(i)
    return print(sum(ret))

In [13]:
print(multiples_three_five(alist))

234168
None


Answer to #1: 234168

2. Even Fibonacci numbers:

Make a list of fibonacci terms (where the values are less than 4 million).
Find the sum of the even values in the list.

In [14]:
alist=[1,2]
i=2
even_num=[]
while alist[-1] < 3500000:
    alist.append(alist[i-1] + alist[i-2])
    i += 1
print(alist)

[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765, 10946, 17711, 28657, 46368, 75025, 121393, 196418, 317811, 514229, 832040, 1346269, 2178309, 3524578]


In [15]:
for num in alist:
    if num % 2 == 0:
        even_num.append(num)
print(even_num)

[2, 8, 34, 144, 610, 2584, 10946, 46368, 196418, 832040, 3524578]


In [16]:
sum(even_num)

4613732

In [17]:
alist=[1,2]
i=2
sums=[]
while alist[-1] < 3500000:
    alist.append(alist[i-1] + alist[i-2])
    i += 1
for num in alist:
    if num % 2 == 0:
        sums.append(num)
print(sum(sums))

4613732


Answer to #2: 4613732

3. largest prime factor number of 600851475143?

In [18]:
#I want to go through the new_list and multiply them until result = 13195.  I want to append those numbers to 
#list to find the max value.
#so I want prime_factors=[1,5,7,13,29]

In [19]:
x=np.arange(1,13196)
new_list=[]
result=1
prime_factors=[]
num=13195
for i in x:
    if num % i == 0:
        new_list.append(i)
        num == num / i

In [20]:
for a in new_list:
    result=result*a
    prime_factors.append(a)
    if result == 13195:
        break
print(max(prime_factors))

29


In [21]:
# y=np.arange(1,600851475144)
# new_list2=[]
# result2=1
# prime_factors2=[]
# num=600851475143
# for i in y:
#     if num % i == 0:
#         new_list2.append(i)
#         num == num / i

In [22]:
# for b in new_list2:
#     result2=result2*a
#     prime_factors2.append(b)
#     if result2 == 13195:
#         break
# print(max(prime_factors2))

Answer to #3: my kernel keeps dying, but the code to get the answer is as above.

4. find the largest palindrome product

In [23]:
a=list(np.arange(100,1000))    #3 digit numbers
b=list(np.arange(100,1000))   #3 digit numbers

#for palindrome product, reading the number forward == reading the number backward.  
#so find values from a and b where their product is the final_value and it equals final_value[::-1].
#print the max value of their product:

In [24]:
for i, j in zip(a,b):
    combination={}
    location=0
    if str(i*j) == str(i*j)[::-1]: 
        location += 1
        combination[location] = str(i*j)
    print(combination)

{}
{1: '10201'}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{1: '12321'}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{1: '14641'}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{1: '40804'}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{1: '44944'}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{1: '69696'}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{1: '94249'}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}
{}


In [26]:
for i, j in zip(a,a):
    combination=0
    product=i * j
    if str(product) == str(product)[::-1]: 
        if product > combination:
            combination = product
        print(combination)

10201
12321
14641
40804
44944
69696
94249
698896


In [27]:
print (max(i*j for i in np.arange(1,1000) for j in np.arange(1,1000) if str(i*j) == str(i*j)[::-1]))

906609


Answer to #4: 906609

5. smallest multiple:

2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.

What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?

for (i) in a(a list of numbers),

divide a[i] by every number in b.

If sum of all their remainder == 0,

break and return a[i]


Not sure how to solve this problem.  Will discuss with my mentor.

In [229]:
from fractions import gcd

print(reduce(lambda a, b: a * b / gcd(a, b), range(1, 21)))

232792560.0


/Users/laurakim/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: DeprecationWarning: fractions.gcd() is deprecated. Use math.gcd() instead.
  This is separate from the ipykernel package so we can avoid doing imports until


6. Sum square difference:

The sum of the squares of the first ten natural numbers is,

12 + 22 + ... + 102 = 385
The square of the sum of the first ten natural numbers is,

(1 + 2 + ... + 10)2 = 552 = 3025
Hence the difference between the sum of the squares of the first ten natural numbers and the square of the sum is 3025 − 385 = 2640.

Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum.

In [6]:
a=list(np.arange(1,101))  
first_list=[]
second_list=[]

In [8]:
def sum_square(alist):
    for i in alist:
        x=i**2
        first_list.append(x)
    return (sum(first_list))


def square_sum(alist):
        y=0
        for i in alist:
            y += i
        return (y**2)

def difference(alist):
        result = square_sum(alist) - sum_square(alist)
        return result

In [9]:
print(difference(a))

25164150


Answer to #6: 25164150

7. 10001st prime:
    
By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.

What is the 10 001st prime number?

Write a code that creates a list of 10001 numbers of the first prime numbers.  Return the last value of the list.

In [151]:
original_list=[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15]
new_list=[]

In [152]:
for i in original_list:
    x=np.arange(2,i)
    a=i % x
    if a.all() != 0:
        new_list.append(i)
        if len(new_list) == 7:
            break
print(new_list[6])

13


In [187]:
def isPrime(n):
    if n < 2: return "Neither prime, nor composite"
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

def nthPrime(n):
    numberOfPrimes = 0
    prime = 1

    while numberOfPrimes < n:
        prime += 1
        if isPrime(prime):
            numberOfPrimes += 1
    return prime

print(nthPrime(10001))

104743


Answer to #7: 104743

8. 
The four adjacent digits in the 1000-digit number that have the greatest product are 9 × 9 × 8 × 9 = 5832.

Find the thirteen adjacent digits in the 1000-digit number that have the greatest product. What is the value of this product?

new line after every 50 digits.
from each row, get a group of 4 numbers.  Return the group with max number of their products.

Or are they just one long list of numbers where I don't have to make a new line after every 50 digits???



In [28]:
w=[1,2,3,4,5]
start=0
end=4
y=[]

In [29]:
while end <= len(w):
        x=w[start:end]
        y.append(np.prod(x))
        start += 1
        end += 1
print (max(y))

120


In [30]:
w=[7,3,1,6,7,1,7,6,5,3,1,3,3,0,6,2,4,9,1,9,2,2,5,1,1,9,6,7,4,4,2,6,5,7,4,7,4,2,3,5,5,3,4,9,1,9,4,9,3,4,9,6,9,8,3,5,2,0,3,1,2,7,7,4,5,0,6,3,2,6,2,3,9,5,7,8,3,1,8,0,1,6,9,8,4,8,0,1,8,6,9,4,7,8,8,5,1,8,4,3,8,5,8,6,1,5,6,0,7,8,9,1,1,2,9,4,9,4,9,5,4,5,9,5,0,1,7,3,7,9,5,8,3,3,1,9,5,2,8,5,3,2,0,8,8,0,5,5,1,1,1,2,5,4,0,6,9,8,7,4,7,1,5,8,5,2,3,8,6,3,0,5,0,7,1,5,6,9,3,2,9,0,9,6,3,2,9,5,2,2,7,4,4,3,0,4,3,5,5,7,6,6,8,9,6,6,4,8,9,5,0,4,4,5,2,4,4,5,2,3,1,6,1,7,3,1,8,5,6,4,0,3,0,9,8,7,1,1,1,2,1,7,2,2,3,8,3,1,1,3,6,2,2,2,9,8,9,3,4,2,3,3,8,0,3,0,8,1,3,5,3,3,6,2,7,6,6,1,4,2,8,2,8,0,6,4,4,4,4,8,6,6,4,5,2,3,8,7,4,9,3,0,3,5,8,9,0,7,2,9,6,2,9,0,4,9,1,5,6,0,4,4,0,7,7,2,3,9,0,7,1,3,8,1,0,5,1,5,8,5,9,3,0,7,9,6,0,8,6,6,7,0,1,7,2,4,2,7,1,2,1,8,8,3,9,9,8,7,9,7,9,0,8,7,9,2,2,7,4,9,2,1,9,0,1,6,9,9,7,2,0,8,8,8,0,9,3,7,7,6,6,5,7,2,7,3,3,3,0,0,1,0,5,3,3,6,7,8,8,1,2,2,0,2,3,5,4,2,1,8,0,9,7,5,1,2,5,4,5,4,0,5,9,4,7,5,2,2,4,3,5,2,5,8,4,9,0,7,7,1,1,6,7,0,5,5,6,0,1,3,6,0,4,8,3,9,5,8,6,4,4,6,7,0,6,3,2,4,4,1,5,7,2,2,1,5,5,3,9,7,5,3,6,9,7,8,1,7,9,7,7,8,4,6,1,7,4,0,6,4,9,5,5,1,4,9,2,9,0,8,6,2,5,6,9,3,2,1,9,7,8,4,6,8,6,2,2,4,8,2,8,3,9,7,2,2,4,1,3,7,5,6,5,7,0,5,6,0,5,7,4,9,0,2,6,1,4,0,7,9,7,2,9,6,8,6,5,2,4,1,4,5,3,5,1,0,0,4,7,4,8,2,1,6,6,3,7,0,4,8,4,4,0,3,1,9,9,8,9,0,0,0,8,8,9,5,2,4,3,4,5,0,6,5,8,5,4,1,2,2,7,5,8,8,6,6,6,8,8,1,1,6,4,2,7,1,7,1,4,7,9,9,2,4,4,4,2,9,2,8,2,3,0,8,6,3,4,6,5,6,7,4,8,1,3,9,1,9,1,2,3,1,6,2,8,2,4,5,8,6,1,7,8,6,6,4,5,8,3,5,9,1,2,4,5,6,6,5,2,9,4,7,6,5,4,5,6,8,2,8,4,8,9,1,2,8,8,3,1,4,2,6,0,7,6,9,0,0,4,2,2,4,2,1,9,0,2,2,6,7,1,0,5,5,6,2,6,3,2,1,1,1,1,1,0,9,3,7,0,5,4,4,2,1,7,5,0,6,9,4,1,6,5,8,9,6,0,4,0,8,0,7,1,9,8,4,0,3,8,5,0,9,6,2,4,5,5,4,4,4,3,6,2,9,8,1,2,3,0,9,8,7,8,7,9,9,2,7,2,4,4,2,8,4,9,0,9,1,8,8,8,4,5,8,0,1,5,6,1,6,6,0,9,7,9,1,9,1,3,3,8,7,5,4,9,9,2,0,0,5,2,4,0,6,3,6,8,9,9,1,2,5,6,0,7,1,7,6,0,6,0,5,8,8,6,1,1,6,4,6,7,1,0,9,4,0,5,0,7,7,5,4,1,0,0,2,2,5,6,9,8,3,1,5,5,2,0,0,0,5,5,9,3,5,7,2,9,7,2,5,7,1,6,3,6,2,6,9,5,6,1,8,8,2,6,7,0,4,2,8,2,5,2,4,8,3,6,0,0,8,2,3,2,5,7,5,3,0,4,2,0,7,5,2,9,6,3,4,5,0]
start=0
end=13
y=[]

In [31]:
while end <= len(w):
        x=w[start:end]
        y.append(np.prod(x))
        start += 1
        end += 1
print (max(y))

23514624000


Answer to #8: 23514624000 

9.  Special Pythagorean triplet:

In [102]:
for a in np.arange(1,1000):
        for b in np.arange(1,1000):
            c = 1000-a-b
            if (a**2+b**2) == c**2:
                print(a*b*c)

31875000
31875000


Answer to #9: 31875000

10. summation of primes

The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.

In [58]:
original_list=[2,3,4,5,6,7,8,9]
new_list=[]

In [59]:
for i in original_list:
    x=np.arange(2,i)
    a=i % x
    if a.all() != 0:
        new_list.append(i)
print(sum(new_list))

17


In [73]:
def sumPrimes(n):
    sum, sieve = 0, [True] * n
    for i in range(2, n):
        if sieve[i]:
            sum += i
            for i in range(i*i, n, i):
                sieve[i] = False
    return sum

print (sumPrimes(2000000))

142913828922


Answer to #10: 142913828922